stan/math/prim/scal/prob/gumbel_rng.hpp

```diff
@@ -3,50 +3,64 @@
 
 #include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
 #include <stan/math/prim/scal/err/check_finite.hpp>
-#include <stan/math/prim/scal/err/check_not_nan.hpp>
-#include <stan/math/prim/scal/err/check_positive.hpp>
-#include <stan/math/prim/scal/meta/length.hpp>
+#include <stan/math/prim/scal/err/check_positive_finite.hpp>
+#include <stan/math/prim/scal/meta/max_size.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
 #include <stan/math/prim/scal/meta/VectorBuilder.hpp>
-#include <stan/math/prim/scal/meta/return_type.hpp>
-#include <stan/math/prim/scal/fun/constants.hpp>
-#include <stan/math/prim/scal/meta/include_summand.hpp>
-#include <stan/math/prim/scal/fun/value_of.hpp>
 #include <boost/random/uniform_01.hpp>
 #include <boost/random/variate_generator.hpp>
 #include <string>
 
 namespace stan {
   namespace math {
-
     /**
      * Return a pseudorandom Gumbel variate with the given location and scale
      * using the specified random number generator.
      *
+     * mu and sigma can each be either scalars or vectors. All vector inputs
+     * must be the same length.
+     *
+     * @tparam T_loc Type of location parameter
+     * @tparam T_scale Type of scale parameter
      * @tparam RNG type of random number generator
-     * @param mu location parameter
-     * @param beta positive scale parameter
+     * @param mu (Sequence of) location parameter(s)
+     * @param beta (Sequence of) scale parameter(s)
      * @param rng random number generator
      * @return Gumbel random variate
      * @throw std::domain_error if mu is infinite or beta is nonpositive.
+     * @throw std::invalid_argument if vector arguments are not the same length
      */
-    template <class RNG>
-    inline double
-    gumbel_rng(double mu,
-               double beta,
+    template <typename T_loc, typename T_scale, class RNG>
+    inline typename VectorBuilder<true, double, T_loc, T_scale>::type
+    gumbel_rng(const T_loc& mu,
+               const T_scale& beta,
```

Comment:
[optional fix]

pack as many arguments onto a line as possible.  Remember, vertical space is precious.  (again, in Google style guide, not in cpplint)